env/env.go

```diff
@@ -0,0 +1,141 @@
+package env
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strconv"
+)
+
+// Lookup must return the value for the given name and whether or not it was
+// found.
+type Lookup func(name string) (string, bool)
+
+// VarSet provides methods to access environment variables from a list of Lookup
+// sources.
+type VarSet struct {
+	// sources are searched sequentially for a given environment variable.
+	sources []Lookup
+}
+
+// NewVarSet makes a new VarSet from opts.
+func NewVarSet(opts ...VarSetOpt) *VarSet {
+	v := &VarSet{sources: make([]Lookup, 0)}
+	for _, opt := range opts {
+		opt(v)
+	}
+	return v
+}
+
+// AddSource adds s as another lookup source.
+func (v *VarSet) AddSource(s Lookup) *VarSet {
+	v.sources = append(v.sources, s)
+	return v
+}
+
+// String gets the string environment variable with the given name, if set.
+// If not found, returns false.
+func (v *VarSet) String(name, defaultVal string) string {
+	rv, ok := v.load(name)
+	if !ok {
+		return defaultVal
+	}
+	return rv
+}
+
+// MustString gets the string environment variable with the given name, if set.
+// It will panic with an error if it is not set.
+// Callers can use IsVarNotFound to check the error.
+func (v *VarSet) MustString(name string) string {
+	rv, ok := v.load(name)
+	if !ok {
+		panic(&varNotFoundError{name: name})
+	}
+	return rv
+}
+
+// Bool gets the boolean environment variable with the given name, if set.
+// If not found, returns false.
+// If found and cannot parse, returns an error.
+func (v *VarSet) Bool(name string) (bool, error) {
+	val, ok := v.load(name)
+	if !ok {
+		return false, nil
+	}
+	rv, err := strconv.ParseBool(val)
+	if err != nil {
+		return false, err
+	}
+	return rv, nil
+}
+
+// MustBool gets the boolean environment variable with the given name, if set.
+// It will panic with an error if it is not set or is not a valid boolean.
+// Callers can use IsVarNotFound to check the error.
+func (v *VarSet) MustBool(name string) bool {
+	val, ok := v.load(name)
+	if !ok {
+		panic(&varNotFoundError{name: name})
+	}
+	rv, err := strconv.ParseBool(val)
+	if err != nil {
+		panic(&varNotFoundError{name: name})
+	}
+	return rv
+}
+
+// MustHexEncodedByteArray gets the hex-encoded environment variable with the
+// given name, if set.
+// It will panic with an error if it is not set if it is not a valid hex-encoded
+// string or if its length does not match decodedByteLength.
+// Callers can use IsVarNotFound to check the error.
```

Comment:
No they can't - we're not returning an error!

Reply:
Oh, yeah they can't use it using the return value since there is none. However they can using `recover` (which they actually do, so I thought it was worth mentioning): https://github.com/18F/cg-dashboard/pull/1258/files#diff-661525df772e761479305d75690a20adR103

Reply:
I clarified the comment in the code. Actually, as a 3rd question for you: I don't like that the package returns `varNotFoundError` for things that failed to parse. Does it sound OK to you to add another error for the failed parsing? I can honestly see callers using this, setting `MY_BOOL=foo` and getting an error `"sorry, expected MY_BOOL to exist but it didn't"` and then just pulling their hair out.